client/src/components/Comparison/HomeInfo/index.js

```diff
@@ -4,7 +4,7 @@ import './index.css';
 import { Link } from 'react-router-dom';
 
 const HomeInfo = props => {
-  const { arrayOfCharity } = props;
+  const { arrayOfCharity, history } = props;
```

Comment:
@othman-shamla You can try using react-router's [withRouter](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/withRouter.md) HOC in this component to have `history` added to props without you having to pass history as a prop yourself (in the Comparison component, line 71).

To use it, you just export this HomeInfo component wrapped in the withRouter function.  So at the bottom of this file, it would become `export default withRouter(HomeInfo);`.

I've found that this cleans up the code a lot, as then you don't have to pass history all over the place yourself (before I used withRouter, sometimes I'd have to pass history through a series of 3 or 4 intermediate components just to use it in the bottom-level component).